week-10-angular/day-02/module-01/readme.md

```diff
@@ -1 +1,223 @@
+
+6. `app.js`
+
+    ```js
+    var app = angular.module("ngFun", []);
+    ```
+
+6. Now update your `body` element to `<body ng-app="ngFun">`.
+
+Sweet, we're up and running!
+
+## Templates & Expressions
+
+Angular creates it's views by templating directly into HTML with expressions. This is it's declarative way of building the UI.
+
+Try writing any regular javascript expression in side double curly brackets, such as: `{{ <someExpression> }}` and see what your HTML evaluates to. What happens what you express:
+
+* `4 * 4`
+* `"hola!".toUpperCase()`
+* `['s','w','e','e','t','n','e','s','s'].join("")`
+
+## Controllers & Scope
+
+Controllers contain all the business logic for our application.
+
+We can seed our application with some data, but first we have to create a controller.
+
+`app.js`
+
+```js
+app.controller("PokemonCtrl", function() {
+	//logic here
+});
+```
+
+Most applications will have several controllers that map to a particular resource. In this case we're using Pokemon.
+
+To use our controller in our View we have to declare it somewhere. Create a new `div` tag that will house our Pokemon Controller.
+
+`index.html`
+
+```html
+<div ng-controller="PokemonCtrl">
+	<!--placeholder for now-->
+</div>
+```
+
+In order to pass data or behavior to our HTML view we need to use the object `$scope`. It is the interface to pass data and behavior into our views. Both the View and Controller share access to the $scope object.
+
+![scope](http://devgirl.org/wp-content/uploads/2013/03/concepts-controller.png)
+
+Let's register some Pokemon with $scope!
+
+```js
+app.controller("PokemonCtrl", function($scope){
+  $scope.pokemon = [
+    {
+      Ndex: 25,
+      name: 'Pikachu',
+      type: 'Electric'
+    },
+    {
+      Ndex: 10,
+      name: 'Caterpie',
+      type: 'Bug'
+    },
+    {
+      Ndex: 39,
+      name: 'Jigglypuff',
+      type: 'Fairy'
+    },
+    {
+      Ndex: 94,
+       name: 'Gengar',
+      type: 'Ghost'
+    },
+    {
+      Ndex: 143,
+      name: 'Snorlax',
+      type: 'Normal'
+    }
+  ];
+});
+```
+
+Great, now let's see if we can see them in our view by referencing the `pokemon` variable inside an expression and wrapping it in a `pre` tag.
+
+```html
+<div ng-controller="PokemonCtrl">
+	<pre>{{ pokemon }}</pre>
+</div>
+```
+
+That's cool, but it doesn't look very great. What if we could format our data so that the View knows to render it as JSON?
+
+### Challenge
+
+* Use an Angular <a href="https://docs.angularjs.org/guide/filter" target="_blank">filter</a> to render the data as JSON! Here are a <a href="https://docs.angularjs.org/api/ng/filter" target="_blank">list</a> of options you can implement.
+
+* Pass a new variable `catchphrase` from the Controller to the View. Set it's value as "gotta catch 'em all!" and use an angular filter to uppercase it in the View.
+
+
+## <a href="https://docs.angularjs.org/guide/directive#what-are-directives-" target="_blank">Directives</a>
```

Comment:
My lesson notes for the afternoon basically start here. Maybe take this out? I think there is a lot to focus on without going too much into directives. I would say focus on setup, templating, and two-way data binding.